### DISCUSSION TITLE
Error that I have no idea how to solve. Please help.

### DISCUSSION THREAD

**Discussion by 0Andras** [Serene-Arc/bulk-downloader-for-reddit#540] [General]
**I've tried to look everywhere but haven't seen this problem appear at all. Any help?**

C:\WINDOWS\system32>py -m pip install bdfr --upgrade

// Note: happens no matter what the directory is here //

Collecting bdfr
  Using cached bdfr-2.4.2-py3-none-any.whl (54 kB)
Requirement already satisfied: click>=7.1.2 in c:\users\USERNAME\appdata\local\programs\python\python310\lib\site-packages (from bdfr) (8.0.3)
Requirement already satisfied: appdirs>=1.4.4 in c:\users\USERNAME\appdata\local\programs\python\python310\lib\site-packages (from bdfr) (1.4.4)
Requirement already satisfied: ffmpeg-python>=0.2.0 in c:\users\USERNAME\appdata\local\programs\python\python310\lib\site-packages (from bdfr) (0.2.0)
Requirement already satisfied: pyyaml>=5.4.1 in c:\users\USERNAME\appdata\local\programs\python\python310\lib\site-packages (from bdfr) (6.0)
Requirement already satisfied: praw>=7.2.0 in c:\users\USERNAME\appdata\local\programs\python\python310\lib\site-packages (from bdfr) (7.4.0)
Requirement already satisfied: bs4>=0.0.1 in c:\users\USERNAME\appdata\local\programs\python\python310\lib\site-packages (from bdfr) (0.0.1)
Requirement already satisfied: yt-dlp>=2021.9.25 in c:\users\USERNAME\appdata\local\programs\python\python310\lib\site-packages (from bdfr) (2021.10.10)
Collecting dict2xml>=1.7.0
  Using cached dict2xml-1.7.0.tar.gz (6.4 kB)
  Installing build dependencies ... done
  Getting requirements to build wheel ... done
  Preparing wheel metadata (pyproject.toml) ... error
  ERROR: Command errored out with exit status 1:
   command: 'C:\Users\USERNAME\AppData\Local\Programs\Python\Python310\python.exe' 'C:\Users\USERNAME\AppData\Local\Programs\Python\Python310\lib\site-packages\pip\_vendor\pep517\in_process\_in_process.py' prepare_metadata_for_build_wheel 'C:\Users\USERNAME\AppData\Local\Temp\tmphxu05542'
       cwd: C:\Users\USERNAME\AppData\Local\Temp\pip-install-d6hee48c\dict2xml_080d4c15d3074545a2de5a98df972422
  Complete output (15 lines):
  Error in sitecustomize; set PYTHONVERBOSE for traceback:
  SyntaxError: (unicode error) 'utf-8' codec can't decode byte 0xe4 in position 0: unexpected end of data (sitecustomize.py, line 7)
  running dist_info
  creating C:\Users\USERNAME\AppData\Local\Temp\pip-modern-metadata-_5ej_qqw\dict2xml.egg-info
  writing C:\Users\USERNAME\AppData\Local\Temp\pip-modern-metadata-_5ej_qqw\dict2xml.egg-info\PKG-INFO
  writing dependency_links to C:\Users\USERNAME\AppData\Local\Temp\pip-modern-metadata-_5ej_qqw\dict2xml.egg-info\dependency_links.txt
  writing requirements to C:\Users\USERNAME\AppData\Local\Temp\pip-modern-metadata-_5ej_qqw\dict2xml.egg-info\requires.txt
  writing top-level names to C:\Users\USERNAME\AppData\Local\Temp\pip-modern-metadata-_5ej_qqw\dict2xml.egg-info\top_level.txt
  writing manifest file 'C:\Users\USERNAME\AppData\Local\Temp\pip-modern-metadata-_5ej_qqw\dict2xml.egg-info\SOURCES.txt'
  reading manifest file 'C:\Users\USERNAME\AppData\Local\Temp\pip-modern-metadata-_5ej_qqw\dict2xml.egg-info\SOURCES.txt'
  reading manifest template 'MANIFEST.in'
  adding license file 'LICENSE'
  writing manifest file 'C:\Users\USERNAME\AppData\Local\Temp\pip-modern-metadata-_5ej_qqw\dict2xml.egg-info\SOURCES.txt'
  creating 'C:\Users\USERNAME\AppData\Local\Temp\pip-modern-metadata-_5ej_qqw\dict2xml.dist-info'
  error: invalid command 'bdist_wheel'
 
WARNING: Discarding https://files.pythonhosted.org/packages/6b/df/5afbfa406632dd54935ea86e7ff31c0e8fb657d13a89a6e6eb53f5a3db79/dict2xml-1.7.0.tar.gz#sha256=1446e7d89b39630e53c20eb38e05046216e3dd72a822954fe7548fa59145d62d (from https://pypi.org/simple/dict2xml/). Command errored out with exit status 1: 'C:\Users\USERNAME\AppData\Local\Programs\Python\Python310\python.exe' 'C:\Users\USERNAME\AppData\Local\Programs\Python\Python310\lib\site-packages\pip\_vendor\pep517\in_process\_in_process.py' prepare_metadata_for_build_wheel 'C:\Users\USERNAME\AppData\Local\Temp\tmphxu05542' Check the logs for full command output.
Collecting bdfr
  Using cached bdfr-2.4.1-py3-none-any.whl (54 kB)
Collecting youtube-dl>=2021.3.14
  Using cached youtube_dl-2021.6.6-py2.py3-none-any.whl (1.9 MB)
Collecting bdfr
  Using cached bdfr-2.4.0-py3-none-any.whl (53 kB)
  Using cached bdfr-2.3.0-py3-none-any.whl (52 kB)
  Using cached bdfr-2.2.0-py3-none-any.whl (52 kB)
  Using cached bdfr-2.1.1-py3-none-any.whl (49 kB)
  Using cached bdfr-2.1.0-py3-none-any.whl (49 kB)
  Using cached bdfr-2.0.0-py3-none-any.whl (70 kB)
ERROR: Cannot install bdfr==2.0.0, bdfr==2.1.0, bdfr==2.1.1, bdfr==2.2.0, bdfr==2.3.0, bdfr==2.4.0, bdfr==2.4.1 and bdfr==2.4.2 because these package versions have conflicting dependencies.

The conflict is caused by:
    bdfr 2.4.2 depends on dict2xml>=1.7.0
    bdfr 2.4.1 depends on dict2xml>=1.7.0
    bdfr 2.4.0 depends on dict2xml>=1.7.0
    bdfr 2.3.0 depends on dict2xml>=1.7.0
    bdfr 2.2.0 depends on dict2xml>=1.7.0
    bdfr 2.1.1 depends on dict2xml>=1.7.0
    bdfr 2.1.0 depends on dict2xml>=1.7.0
    bdfr 2.0.0 depends on dict2xml>=1.7.0

To fix this you could try to:
1. loosen the range of package versions you've specified
2. remove package versions to allow pip attempt to solve the dependency conflict

ERROR: ResolutionImpossible: for help visit https://pip.pypa.io/en/latest/user_guide/#fixing-conflicting-dependencies

**I'm not sure what is causing this or how it could be fixed, but if anybody has any ideas please let me know. I'm not good at coding and such so expect me to be an idiot when it comes to these things. I did download everything that was in the requirements, or at least I think so.
Also I'm hoping this is the right place to post this.**

**Comment by Serene-Arc** [comment]
So the actual root error is this:

>   error: invalid command 'bdist_wheel'

You need to install the package wheel, with `python3 -m pip install -U wheel`. See how it goes after that.

**Reply by 0Andras** [reply]
Alright that worked, thank you so much.
